databroker/_drivers/mongo_normalized.py

```diff
@@ -283,11 +279,21 @@ def _get_resources(self, run_start_uid):
             {'run_start': run_start_uid}, {'_id': False}))
 
     def _get_resource(self, uid):
-        doc = self._resource_collection.find_one(
-            {'uid': uid})
+        doc = self._resource_collection.find_one({'uid': uid}, {'_id': False})
+
+        # Some old resource documents don't have a 'uid' and they are
+        # referenced by '_id'.
+        if doc is None:
+            try:
+                uid = ObjectId(uid)
```

Comment:
Redefining local variables is something I try to avoid. It can make code harder to read, and it can reduce the usefulness of interactive debugging  (i.e. `%debug`) because you can't access the original value.
```suggestion
                _id = ObjectId(uid)
```